subset-routing-test/src/main/java/com/hazelcast/jet/tests/client/subset/routing/SubsetRoutingTest.java

```diff
@@ -0,0 +1,202 @@
+                SLEEP_BETWEEN_CONNECTION_ASSERTION_ATTEMPTS_IN_SECONDS);
+    }
+
+    @Override
+    protected void beforeTest(HazelcastInstance client, ClusterType clusterType) throws Exception {
+    }
+
+    @Override
+    protected void test(HazelcastInstance client, ClusterType clusterType, String name) throws Throwable {
+        long begin = System.currentTimeMillis();
+        String logPrefix = "[" + name + "] ";
+
+        HazelcastClientInstanceImpl clientInstance = getHazelcastClientInstanceImpl(client);
+
+        long validationCount = 0;
+
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            assertEffectiveMembersAndActiveConnections(clientInstance);
+            assertRequestToCluster(clientInstance);
+
+            validationCount++;
+            if (validationCount % logVerificationCountThreshold == 0) {
+                logger.info(logPrefix + "Validations count: " + validationCount);
+            }
+            sleepMinutes(sleepBeforeValidationInMinutes);
+
+        }
+        logger.info(logPrefix + "Final validations count: " + validationCount);
+    }
+
+    private void assertRequestToCluster(HazelcastClientInstanceImpl clientInstance) {
+        AtomicReference<Throwable> lastException = new AtomicReference<>();
+
+        ConditionVerifierWithTimeout verifier = new ConditionVerifierWithTimeout(
+                Duration.ofMinutes(maxWaitForExpectedConnectionInMinutes),
+                Duration.ofSeconds(sleepBetweenConnectionAssertionAttemptsInSeconds));
+
+        verifier
+                //This method throws OperationTimeoutException for unavailable cluster
+                .condition(() -> clientInstance.getMap(this.getClass().getSimpleName()).isEmpty())
+                .onException(lastException::set)
+                .onTimeout(() -> {
+                    if (lastException.get() != null) {
+                        throw new IllegalStateException("Timed out waiting for cluster active connection",
+                                lastException.get());
+                    }
+                    throw new IllegalStateException("Timed out waiting for cluster active connection");
+                })
+                .execute();
+
+    }
+
+    @Override
+    protected List<ClusterType> runOnClusters() {
+        return List.of(DYNAMIC, STABLE);
+    }
+
+    private void assertEffectiveMembersAndActiveConnections(HazelcastClientInstanceImpl clientInstance) {
+        Duration maxLoopDuration = Duration.ofMinutes(maxWaitForExpectedConnectionInMinutes);
+        Duration sleepBetweenAttempts = Duration.ofSeconds(sleepBetweenConnectionAssertionAttemptsInSeconds);
+
+        ConditionVerifierWithTimeout verifier = new ConditionVerifierWithTimeout(maxLoopDuration, sleepBetweenAttempts);
+
+        verifier.condition(() -> {
+                    Collection<Member> effectiveMemberList = getEffectiveMemberList(clientInstance);
```

Comment:
The result of `getEffectiveMemberList` can be different in `condition` and in the `onCondition*` methods.

Reply:
I was aware, but I didn't want to complicate the code, but introduced now a context for `ConditionVerifier` :
```
                    return new ConditionResult<>(
                            effectiveMemberList.size() == 1 && activeConnections == 1,
                            new ConnectionsContext(activeConnections, effectiveMemberList));
```
This context is passed into pass/fail  methods now.